### ISSUE TITLE
Extent this to widgets

### ISSUE THREAD

**Issue by pbearne** [wp-shortcake/shortcake#210]
Hi Guys 

This more of an idea than an issue

In my plugin author avatars list I provide an TinyMCE plugin with a form in popup to generate a short-code and use the same form in a widget so it can be every page.

I am looking at replacing tinyMCE plugin code with this but that will still leave me with the widget code to maintain.

So to the idea how hard would it be to wrap the generated from into a widget control 

The dirty way to (simplest) would to yes save the created short-code and pass it through do_shortocde() on the way out

I guess that we would need a flag to turn this on and off

Any thoughts?

Would you take a pull request for this?

Paul

**Comment by danielbachhuber** [comment]
Related #153

It would be neat if the "Insert Post Element" interface could be used independently of the shortcode preview (and vice versa).

> Would you take a pull request for this?

I'm open to a pull request that attempts this decoupling behavior :) Now that our JS has been abstracted (#142), it should be easier to do

**Comment by szepeviktor** [comment]
@danielbachhuber Is this the issue about registering a shortcode as a widget? And of course editing it by Shortcake UI.

**Comment by danielbachhuber** [comment]
> Is this the issue about registering a shortcode as a widget? And of course editing it by Shortcake UI.

No, the opposite in fact: being able to use shortcodes in widgets.

**Comment by szepeviktor** [comment]
Thank you.

> No, the opposite in fact: being able to use shortcodes in widgets.

So Shortcake UI would appear in Appearance/Widgets?

**Comment by danielbachhuber** [comment]
> So Shortcake UI would appear in Appearance/Widgets?

More or less. Implementation TBD.

**Comment by szepeviktor** [comment]
Sounds exciting.

**Comment by westonruter** [comment]
See also #585 and https://core.trac.wordpress.org/ticket/33473
